### PR TITLE
docs: Cynthion documentation branch

### PR DESCRIPTION
**Pull request by antoinevg**
This is the Cynthion documentation branch.

### CODE REVIEW

**Review by mossmann — CHANGES_REQUESTED** [review]
Great job! I'm very excited to get this cleaned up and merged. I think it is close, but I would like to see several changes. Perhaps the most important change is that the two photographs could be much smaller. I would rather not have giant images taking up room in the repo forever. Also, can the photographs be JPEGs instead of PNGs?

File: docs/source/introduction.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+
+Top View
+~~~~~~~~
+
+.. image:: ../images/cynthion-top.svg
+  :width: 300
+  :alt: Cynthion Top View
+
+- **LEDS A-E** - Five status LEDs managed by the debug micro-contoller.
```
[the remark this answers is not in the document]
Comment (antoinevg) [reply docs/source/introduction.rst:26]: Sticking with the convention for USB ports to match hardware exactly so it's just `A-E` and `0-5` -- doing same for PMOD's. Description text says that it's an LED.

[the remark this answers is not in the document]
Comment (antoinevg) [reply docs/source/introduction.rst:26]: Fixed. Sphinx was being silly. It likes lots of whitespace between things.

File: docs/source/hardware/bringup_guide.rst [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+to move on to the next step.
+
+Optional: Bootloader Locking
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Optionally, you can reversibly lock the bootloader region of the Debug
+Controller, preventing you from accidentally overwriting the bootloader.
+This is most useful for users developing code for the Debug Controller.
+
+If you choose to lock the bootloader, you should lock the first ``4KiB``
```
Comment (martinling) [inline docs/source/hardware/bringup_guide.rst:166]: This should be 2KiB now, right?

Comment (antoinevg) [reply docs/source/hardware/bringup_guide.rst]: 👍 updated!

File: docs/source/developer/soc_firmware_compilation.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+Building and Running
+--------------------
+
+Firmware for the Cynthion SoC can be found in the ``firmware/moondancer/`` sub-directory.
+
+You can rebuild the firmware using ``cargo`` as follows:
+
+.. code-block:: text
+
+    # change to the cynthion firmware directory
```
Comment (claui) [inline docs/source/developer/soc_firmware_compilation.rst:16]: You probably want to capitalize this here?

```suggestion
    # change to the Cynthion firmware directory
```

File: docs/source/developer/soc_firmware_compilation.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    cd firmware/moondancer/
+
+    # rebuild the firmware
+    cargo build --release
+
+To upload the firmware binary to Cynthion and flash the SoC bitstream you can run:
+
+.. code-block:: text
+
+    # change to the cynthion firmware directory
```
Comment (claui) [inline docs/source/developer/soc_firmware_compilation.rst:26]:
```suggestion
    # change to the Cynthion firmware directory
```

File: docs/source/developer/soc_firmware_compilation.rst [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    .. code-block:: sh
+
+        BITSTREAM=path/to/bitstream.bit cargo run --release
+
+
+Running Firmware Unit Tests
+---------------------------
+
+Once the firmware is running on the SoC you can execute some unittests to exercise the firmware.
```
Comment (claui) [inline docs/source/developer/soc_firmware_compilation.rst:55]:
```suggestion
Once the firmware is running on the SoC you can execute some unit tests to exercise the firmware.
```

File: docs/source/developer/soc_firmware_compilation.rst [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+
+Firmware Examples
+-----------------
+
+There are a number of firmware examples in the ``firmware/moondancer/examples/`` sub-directory.
+
+.. code-block:: sh
+
+    # change to the cynthion firmware directory
```
Comment (claui) [inline docs/source/developer/soc_firmware_compilation.rst:76]:
```suggestion
    # change to the Cynthion firmware directory
```

File: docs/source/developer/introduction.rst [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+Installation
+------------
+
+For development you'll need a local copy of the cynthion repository:
+
+.. code-block:: sh
+
+    git clone https://github.com/greatscottgadgets/cynthion.git
+
+To install the ``cynthion`` python package to allow for in-place editing of the sources you can use the ``pip --editable`` command:
```
Comment (claui) [inline docs/source/developer/introduction.rst:38]:
```suggestion
To install the ``cynthion`` Python package to allow for in-place editing of the sources you can use the ``pip --editable`` command:
```

File: docs/source/developer/introduction.rst [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+.. code-block:: sh
+
+    # change to the cynthion python package directory
+    cd cynthion/python/
+
+    # install the cynthion python package, including dependencies required for gateware development
```
Comment (claui) [inline docs/source/developer/introduction.rst:45]:
```suggestion
    # change to the `cynthion` Python package directory
    cd cynthion/python/

    # install the `cynthion` Python package, including dependencies required for gateware development
```

File: docs/source/developer/bitstream_generation.rst [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+- ``soc``      -- System-on-Chip for using Cynthion with Facedancer.
+
+Bitstreams can be generated from the cynthion python package sub-directory as follows:
+
+Analyzer Gateware
+^^^^^^^^^^^^^^^^^
+
+.. code-block:: sh
+
+    # change to the cynthion python package directory
```
[the remark this answers is not in the document]
Comment (antoinevg) [reply docs/source/developer/bitstream_generation.rst:23]: Backticking cynthion won't work in a code block, but you're on for a capital-P Python!

File: docs/source/developer/introduction.rst [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+- `Python <https://wiki.python.org/moin/BeginnersGuide/Download>`__ v3.8, or later.
+- A working FPGA toolchain. We only officially support a toolchain
+  composed of the `Project Trellis <https://github.com/YosysHQ/prjtrellis>`__
+  ECP5 tools, the `yosys <https://github.com/YosysHQ/yosys>`__
+  synthesis suite, and the `NextPNR <https://github.com/YosysHQ/nextpnr>`__
+  place-and-route tool. You can obtain the latest binary distribution of this
+  software from the `oss-cad-suite-build <https://github.com/YosysHQ/oss-cad-suite-build>`__
+  project.
+-  A working installation of
+   `Amaranth HDL <https://github.com/amaranth-lang/amaranth>`__.
```
[the remark this answers is not in the document]
Comment (antoinevg) [reply docs/source/developer/introduction.rst:24]: That shouldn't be in the list at all -- it's already a package dependency, tx!

**Review by claui — APPROVED** [review]
Looks great now!  
(Assuming you’re aware of the remaining `TODO` markers throughout the doc.)

**Review by martinling — APPROVED** [review]
Looks good to me. Obviously there's still things to add but everything here looks fine and we should go ahead and get it in.

**Pull request merged**